dexpv2/segmentation.py

```diff
@@ -86,6 +91,11 @@ def fancy_otsu_threshold(
 
     hist, bin_centers = exposure.histogram(image, nbins)
 
+    # clip bins and histogram beyond max_foreground value
+    if max_foreground != 0.0:
+        hist = [hist[i] for i in range(len(hist)) if bins[i] < max_foreground]
+        bins = [bin for bin in bins if bin < max_foreground]
```

Comment:
```suggestion
    if max_foreground is not None:
        below_threshold_mask = bin < max_foreground
        bins = bins[below_threshold_mask]
        hist = hist[below_threshold_mask]
```

this makes it a bit more pythonic, could you check if I didn't introduce any mistakes?